group-project/src/main/webapp/script.js

```diff
@@ -68,8 +68,28 @@ function initMap(){
         createMarker(location, map);
     }
 
+
+    // Add Autocomplete
+    let autocomplete = new google.maps.places.Autocomplete(document.getElementById('search-input'), {
+        type: ['establishment'],
+        fields: ['geometry', 'name']
+    });
+
+    autocomplete.addListener('place_changed', () => {
+        const place = autocomplete.getPlace();
+        new google.maps.Marker({
+            position: place.geometry.location,
+            label: place.name,
+            map:map
+        })
+    });
+
 }
 
+
```

Comment:
Did you mean to add this whitespace down here?

Reply:
oh I often leave a lot of spaces in my code to space things out. Is that not a good habit to have?

Reply:
It's fine as long as everyone is on board with that convention and it stays consistent. So, like, you can all agree that there are 4 empty lines between methods and everyone follows that convention.

Honestly, one empty line between methods is the standard anywhere I've worked, Google included. Here's an example from Chromium - https://source.chromium.org/chromium/chromium/src/+/main:chrome/android/features/start_surface/java/src/org/chromium/chrome/features/start_surface/TabSwitcherAndStartSurfaceLayout.java

Reply:
Thank you. I think 1 line between methods is good, and I'll be more mindful about it. Good tips btw. 😄

Reply:
Also note that most of us use auto-formatters and linters for our code before checking it in. This can be accomplished in git via pre-commit hooks. Here's an example I found - https://prettier.io/docs/en/precommit.html

Definitely way too much for an SPS project but if you're ever working on larger projects, things like this are a good option for keeping things consistent without having to memorize a bunch of styles and conventions.

Reply:
Thank you for the information. It's definitely good to know about this 👍